src/main/java/mate/academy/springboot/datajpa/controller/ProductController.java

```diff
@@ -0,0 +1,69 @@
+package mate.academy.springboot.datajpa.controller;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import mate.academy.springboot.datajpa.dto.ProductRequestDto;
+import mate.academy.springboot.datajpa.dto.ProductResponseDto;
+import mate.academy.springboot.datajpa.mapper.ProductMapper;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.service.ProductService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+    private final ProductService productService;
+    private final ProductMapper productMapper;
+
+    @Autowired
+    public ProductController(ProductService productService,
+                             ProductMapper productMapper) {
+        this.productService = productService;
+        this.productMapper = productMapper;
+    }
+
+    /*
+    - get all products where price is between two values received as a `RequestParam` inputs
+    - get all products in categories
+    (you should think how you will receive a list of categories as a `RequestParam` input)
+    */
+
+    @PostMapping
+    public ProductResponseDto create(ProductRequestDto requestDto) {
+        Product product = productService.save(productMapper.toModel(requestDto));
+        return productMapper.toDto(product);
+    }
+
+    @GetMapping("/{id}")
+    public ProductResponseDto getById(@PathVariable Long id) {
+        return productMapper.toDto(productService.findById(id));
+    }
+
+    @DeleteMapping("/{id}")
+    public void delete(Long id) {
+        productService.deleteById(id);
+    }
+
+    @PutMapping("/{id}")
+    public ProductResponseDto update(@PathVariable Long id, ProductRequestDto requestDto) {
+        Product product = productMapper.toModel(requestDto);
+        product.setId(id);
+        return productMapper.toDto(productService.save(product));
+    }
+
+    @GetMapping
+    public List<ProductResponseDto> findAllByParameters(@RequestParam Map<String, String> params) {
```

Comment:
why map?
why all these spesification stuff? can you explain why how did you come up with this?